packages/next/src/server/base-server.ts

```diff
@@ -2637,7 +2637,18 @@ export default abstract class Server<
             const parsedInitUrl = parseUrl(
               getRequestMeta(req, 'initURL') || req.url
             )
-            request.url = `${parsedInitUrl.pathname?.replace(/(\.prefetch\.rsc|\.rsc)$/, '')}${parsedInitUrl.search || ''}`
+            let initPathname = parsedInitUrl.pathname || '/'
+
+            for (const normalizer of [
+              this.normalizers.segmentPrefetchRSC,
```

Comment:
Should this enumerate over all possible normalizers in case we add more later?

Reply:
we're gonna be hoisting the normalizers out of here next so ideally we don't need to add more in the mean-time